React/my-app/src/screens/book_detail/index.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import defaultImg from "../../assets/photos/bigDefault.png";
+import Book_detail from "./layout.js";
+import data from "../../config/data.json";
+
+class BookDetailContainer extends Component {
+  state = {
+    author: null,
+    title: null,
+    genre: null,
+    year: null,
+    image: defaultImg
+  };
+
+  componentDidMount() {
+    this.selectBook();
+  }
+
+  selectBook() {
+    const book = data.find(this.sameBook);
+    this.setState({
+      author: book.author,
+      title: book.title,
+      genre: book.genre,
+      year: book.year
+    });
+    if (book.image_url) {
+      this.setState({ image: book.image_url });
```

Comment:
each setState trigger a new render, so we have to try avoid this using just a setState call
you can do 
```
this.setState({
       author: book.author,
       title: book.title,
       genre: book.genre,
       year: book.year,
       image: book.image_url ? book.image_url : default image
     });
```

Reply:
another option is se image with book.image_url
so, if image_url is undefined, you can can pass to the component on the same way, because you should have a defaultProp on the book component.